modules/apps/commerce/commerce-product-test/src/testIntegration/java/com/liferay/commerce/product/service/test/CPConfigurationEntryLocalServiceTest.java

```diff
@@ -134,6 +142,280 @@ public void testAddCPConfigurationEntry() throws Exception {
+					cpConfigurationEntry2.getCPConfigurationEntryId(),
+					CPConfigurationEntrySettingConstants.TYPE_INDEX_IDS);
+
+		Assert.assertNotNull(cpConfigurationEntrySetting);
+
+		Assert.assertTrue(
+			StringUtil.contains(
+				cpConfigurationEntrySetting.getSetting(),
+				String.valueOf(
+					cpConfigurationList3.getCPConfigurationListId())));
+
+		cpConfigurationEntrySetting =
+			_cpConfigurationEntrySettingLocalService.
+				fetchCPConfigurationEntrySetting(
+					cpConfigurationEntry1.getCPConfigurationEntryId(),
+					CPConfigurationEntrySettingConstants.TYPE_INDEX_IDS);
+
+		Assert.assertNotNull(cpConfigurationEntrySetting);
+
+		Assert.assertFalse(
+			StringUtil.contains(
+				cpConfigurationEntrySetting.getSetting(),
+				String.valueOf(
+					cpConfigurationList3.getCPConfigurationListId())));
+
+		_cpConfigurationEntryLocalService.deleteCPConfigurationEntry(
+			cpConfigurationEntry2);
+
+		cpConfigurationEntrySetting =
+			_cpConfigurationEntrySettingLocalService.
+				fetchCPConfigurationEntrySetting(
+					cpConfigurationEntry1.getCPConfigurationEntryId(),
+					CPConfigurationEntrySettingConstants.TYPE_INDEX_IDS);
+
+		Assert.assertNotNull(cpConfigurationEntrySetting);
+
+		Assert.assertTrue(
+			StringUtil.contains(
+				cpConfigurationEntrySetting.getSetting(),
+				String.valueOf(
+					cpConfigurationList2.getCPConfigurationListId())));
+
+		Assert.assertTrue(
+			StringUtil.contains(
+				cpConfigurationEntrySetting.getSetting(),
+				String.valueOf(
+					cpConfigurationList3.getCPConfigurationListId())));
+	}
+
+	@Test
+	public void testInheritCPConfigurationEntry() throws Exception {
+		frutillaRule.scenario(
+			"Inherit Product Configuration Entry"
+		).given(
+			"There is a Commerce Catalog and its master configuration"
+		).when(
+			"A Configuration Entry is added"
+		).and(
+			"The Configuration Entry is created"
+		).and(
+			"A child Configuration List is added"
+		).then(
+			"The Configuration Entry is inherited"
+		);
+
+		String externalReferenceCode = RandomTestUtil.randomString();
+
+		CPConfigurationEntry cpConfigurationEntry =
+			_cpConfigurationEntryLocalService.addCPConfigurationEntry(
+				externalReferenceCode, _user.getUserId(),
+				_cpConfigurationList.getGroupId(),
+				_portal.getClassNameId(CPDefinition.class),
+				_cpDefinition.getCPDefinitionId(),
+				_cpConfigurationList.getCPConfigurationListId(), 0, "123", true,
+				0, "cpde", 1.0, true, true, true, 1.0, "lowstoc",
+				BigDecimal.TEN, BigDecimal.ONE, BigDecimal.ONE, BigDecimal.ONE,
+				true, true, 1.0, true, true, true, 1.0, 1.0);
+
+		Assert.assertNotNull(cpConfigurationEntry);
+		Assert.assertEquals(
+			externalReferenceCode,
+			cpConfigurationEntry.getExternalReferenceCode());
+
+		Date date = new Date();
+
+		Calendar calendar = CalendarFactoryUtil.getCalendar(date.getTime());
+
+		int displayDateHour = calendar.get(Calendar.HOUR);
+
+		if (calendar.get(Calendar.AM_PM) == Calendar.PM) {
+			displayDateHour += 12;
+		}
+
```

Comment:
for this I have a question: why we cannot use Calender.HOUR_OF_DAY to reach the same goal?

Reply:
IDK this is just a copy paste